code/modules/stock_market/stocks.dm

```diff
@@ -139,8 +139,16 @@
 			speculation += rand(-400, 0) / 1000 * speculation
 			if (prob(1) && prob(5)) // pop that bubble
 				speculation += rand(-4000, 0) / 1000 * speculation
-
-	current_value += (speculation / rand(25000, 50000) + performance / rand(100, 800)) * current_value
+	var/fucking_stock_spikes = current_value + 500
+	var/piece_of_shit_fuck = current_value - 500
+	var/i_hate_this_code = (speculation / rand(25000, 50000) + performance / rand(100, 800)) * current_value
+	if(i_hate_this_code > fucking_stock_spikes || i_hate_this_code < piece_of_shit_fuck)
+		message_admins("STOCK ALERT: [name] attempted to increase/decrease its value by more than 500 credits, this has been stopped. Please alert iamgoofball on #coderbus at irc.rizon.net with the following info:")
+		message_admins("STOCK ALERT: Speculation was: [speculation]")
+		message_admins("STOCK ALERT: Performance was: [performance]")
+		message_admins("STOCK ALERT: If you can, take a screenshot of this data.")
```

Comment:
This is going to be obtuse to admins and difficult for you to ever analyze.
It'd be all around better to do what was discussed before: send these to an investigate log that's saved at round-end so the info is both self-contained and easy to interpret.
You may also want to include a timestamp and the stock transaction log made by players.